### PR TITLE
feat: react button component

### PR DESCRIPTION
**Pull request by BogdanDanilescu**
### Implementation of the HTML Button component

Implementation of the Button component which contains 3 variants and each of the variant has 3 appearances. There is also the option the render a icon button by omitting the label prop. There is also the ability to add a disabled state whenever necessary

Docs have also been updated

Props:
```
label
variant
appearance
size
disabled
icon
iconEnd
```
image
image
image
image
image

### CODE REVIEW

File: packages/react/ds/src/button/types.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  Medium = 'medium',
+  Small = 'small',
+  Large = 'large',
+}
+
+export type ButtonProps = {
+  variant?: ButtonVariant;
+  appearance?: ButtonAppearance;
+  size?: ButtonSize;
+  label?: string;
```
Comment (pregno) [inline packages/react/ds/src/button/types.ts:25]: See comment below about icons

File: packages/html/ds/src/button/button.html [hunk 2]
```diff
@@ -8,11 +8,13 @@
 {% set size = props.size or 'medium' %}
 {% set isDisabled = 'disabled' if props.disabled else 'notDisabled' %}
 
-<button data-element="button-container" data-module="gieds-button" data-testid="govieButton-{{ appearance }}-{{ variant }}-{{ size }}-{{ isDisabled }}" class="gi-btn {{ getVariantAppearanceClass(props.disabled, props.variant, props.appearance) | trim }} {{getSizeClass(props.size) | trim }} {{getIconPosition(props.icon.position) | trim }} {{isButtonDisabled(props.disabled, props.variant, props.appearance) | trim }}">
+<button data-element="button-container" data-module="gieds-button" data-testid="govieButton-{{ appearance }}-{{ variant }}-{{ size }}-{{ isDisabled }}" class="gi-btn {{ getVariantAppearanceClass(props.disabled, props.variant, props.appearance) | trim }} {{getSizeClass(props.size) | trim }} {{getIconPosition(props.iconEnd) | trim }} {{isButtonDisabled(props.disabled, props.variant, props.appearance) | trim }}">
   {% if props.icon %}
-    {{ govieIcon(props.icon.props)}}
+    {{ govieIcon(props.icon)}}
+  {% endif %}
+  {% if props.label %}
```
Comment (pregno) [inline packages/html/ds/src/button/button.html:15]: You can apply the same logic here as I suggested in React.
If you have the property `content,` you can pass text and HTML (as we did for containers or paragraphs).
In this way, we can leave the developer to decide how to place the icon and the content.

File: apps/docs/content/3-components/2-library/5-button.mdx [hunk 3]
```diff
@@ -2,13 +2,249 @@
 title: Button
 description: Button HTML Component
 draft: false
-status: in-development
+status: stable
```
Comment (hamza14khan) [inline apps/docs/content/3-components/2-library/5-button.mdx:5]: Shouldn't this be in-development? @pregno

**Review by hamza14khan — CHANGES_REQUESTED** [review]
Looks good. Happy to approve it once Andrea comments are resolved.

**Review by hamza14khan — APPROVED** [review]
Looks good, just one comment left for @pregno

File: apps/docs/content/3-components/2-library/5-button.mdx (hunk 3 above)
Comment (pregno) [reply apps/docs/content/3-components/2-library/5-button.mdx:5]: I think that is the status of the documentation.
Below there is the status of the library and is set to `alpha`

**Review by pregno — APPROVED** [review]

**Pull request merged**